plugins/repository-s3/src/main/java/org/elasticsearch/cloud/aws/AwsSigner.java

```diff
@@ -34,20 +34,29 @@ private AwsSigner() {
      * @param configuration AWS Client configuration
      * @throws IllegalArgumentException if signer does not exist
      */
-    public static void configureSigner(String signer, ClientConfiguration configuration)
+    public static void configureSigner(String signer, ClientConfiguration configuration, String endpoint)
         throws IllegalArgumentException {
 
         if (signer == null) {
             throw new IllegalArgumentException("[null] signer set");
         }
 
-        try {
-            // We check this signer actually exists in AWS SDK
-            // It throws a IllegalArgumentException if not found
-            SignerFactory.getSignerByTypeAndService(signer, null);
-            configuration.setSignerOverride(signer);
-        } catch (IllegalArgumentException e) {
-            throw new IllegalArgumentException("wrong signer set [" + signer + "]");
+        if (signer.equals("S3SignerType")) {
+            // V2 signer type is not support in aws China and Frankfurt region
+            if (endpoint.equals("s3.cn-north-1.amazonaws.com.cn") ||
+                    endpoint.equals("s3.eu-central-1.amazonaws.com")) {
+                throw new IllegalArgumentException(endpoint + " does not support S3SignerType");
+            }
+        } else if (!signer.equals("AWSS3V4SignerType")) {
+            try {
+                // We check this signer actually exists in AWS SDK
+                // It throws a IllegalArgumentException if not found
+                SignerFactory.getSignerByTypeAndService(signer, null);
+            } catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException("wrong signer set for [cloud.aws.s3.signer] or [cloud.aws.signer]: " + signer);
```

Comment:
Should we actually control this? I know, I wrote that code as well at the begining but the more I think about it, the more I think it's not that useful to block...

May be only log.warn with "[{}] signer might not be supported"...